pwiz_tools/Skyline/Model/DdaSearch/MSAmandaSearchWrapper.cs

```diff
@@ -266,7 +266,7 @@ public override bool Run(CancellationTokenSource tokenSource, IProgressStatus st
                         }
                         finally
                         {
-                            SearchEngine?.Dispose();
+                            SearchEngine.Dispose();
```

Comment:
I imagine there's a reason for the change, but on the face of it this seems less cautious than before?

Reply:
We can choose either this, or we can test for null above where it gets used without a null check in the main try block. I suppose we could test for null above and then throw an exception that the SearchEngine was not set. It didn't look to me like the code ever really expected it to be null. Maybe we should ask Matt to weigh in since this is his code.

Reply:
@chambm Can you pay special attention to this one and see what you think? Should we add a test for if (SearchEngine == null) above, or just remove the test for it here?

Reply:
I basically agreed with Brian so I didn't comment earlier. SearchEngine is set in InitializeEngine. If that function throws before SearchEngine is set, it will be null in the finally block. If it throws after SearchEngine is set or doesn't throw at all, it won't be null. And if it doesn't throw at all, the rest of the try block runs and it won't be null so those SearchEngine calls don't need a null check. What am I missing?

Reply:
As Brendan explained it in conversation at my desk, ReSharper notices that we're checking for null in one place (that "?.") but not others nearby, so it barks. Check in neither and it's placated. Sounds like Brendan's change is the right one.

Reply:
No, I'm quite certain I put that nullsafe dereference there in response to seeing a NRE there. To appease ReSharper the simplest thing is to just do nullsafe derefs above as well. ReSharper is being too conservative here but that's par for the course (better safe than sorry). Another way to appease ReSharper that would also make the code clearer would be to return SearchEngine from InitializeEngine so it's clear in the finally block that SearchEngine is set after that line (as long as the function can't return null, which it can't, but I'm not sure if ReSharper can know that without an explicit attribute).

Reply:
```

[NotNull]
        private MSAmandaSearch InitializeEngine(CancellationTokenSource token, string spectrumFileName)
        {
            _outputParameters = new OutputParameters();
            _outputParameters.FastaFiles = FastaFileNames.ToList();
            _outputParameters.DBFile = FastaFileNames[0];
            //2 == mzid
            _outputParameters.SetOutputFileFormat(2);
            _outputParameters.IsPercolatorOutput = true;
            _outputParameters.SpectraFiles = new List<string>() { spectrumFileName};
            Settings.GenerateDecoyDb = true;
            Settings.ConsideredCharges.Clear();
            foreach(var chargeStr in AdditionalSettings[CONSIDERED_CHARGES].Value.ToString().Split(','))
                Settings.ConsideredCharges.Add(Convert.ToInt32(chargeStr));
            Settings.ChemicalData.UseMonoisotopicMass = true;
            Settings.ReportBothBestHitsForTD = false;
            Settings.CombineConsideredCharges = false;
            //Settings.WriteResultsTwice = true;
            //Settings.ForceTargetDecoyMode = false;
            //Console.WriteLine("\nReportBothBestHitsForTD CombineConsideredCharges WriteResultsTwice ForceTargetDecoyMode");
            //Console.WriteLine($@"{Settings.ReportBothBestHitsForTD}       {Settings.CombineConsideredCharges}        {Settings.WriteResultsTwice}       {Settings.ForceTargetDecoyMode}");
            mzID.Settings = Settings;
            var searchEngine = new MSAmandaSearch(helper, _baseDir.DirPath, _outputParameters, Settings, token);
            searchEngine.InitializeOutputMZ(mzID);
            Settings.LoadedProteinsAtOnce = (int) AdditionalSettings[MAX_LOADED_PROTEINS_AT_ONCE].Value;
            Settings.LoadedSpectraAtOnce = (int) AdditionalSettings[MAX_LOADED_SPECTRA_AT_ONCE].Value;
            return searchEngine;
        }


...

SearchEngine = InitializeEngine(tokenSource, rawFileName.GetSampleLocator());
```

Reply:
Okay. I came up with a no-op assignment that silenced ReSharper and which should have no impact on the logic of this code.